integration/log_rotation_test.go

```diff
@@ -0,0 +1,153 @@
+package integration
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/containous/traefik/integration/try"
+	"github.com/go-check/check"
+	checker "github.com/vdemeester/shakers"
+)
+
+// Log rotation integration test suite
+type LogRotationSuite struct{ BaseSuite }
+
+func (s *LogRotationSuite) TestAccessLogRotation(c *check.C) {
+	// Start Traefik
+	cmd, _ := s.cmdTraefik(withConfigFile("fixtures/access_log_config.toml"))
+	err := cmd.Start()
+	c.Assert(err, checker.IsNil)
+	defer cmd.Process.Kill()
+	defer os.Remove(traefikTestAccessLogFile)
+	defer os.Remove(traefikTestLogFile)
+
+	// Verify Traefik started ok
+	verifyEmptyErrorLog(c, "traefik.log")
+
+	// Start test servers
+	ts1 := startAccessLogServer(8081)
+	defer ts1.Close()
+	ts2 := startAccessLogServer(8082)
+	defer ts2.Close()
+	ts3 := startAccessLogServer(8083)
+	defer ts3.Close()
+
+	// Allow time to startup
+	time.Sleep(1 * time.Second)
+
+	// Make some requests
+	_, err = http.Get("http://127.0.0.1:8000/test1")
+	c.Assert(err, checker.IsNil)
+	// in the midst of the requests, issue SIGUSR1 signal to server process and move access log
+	err = cmd.Process.Signal(syscall.SIGUSR1)
+	c.Assert(err, checker.IsNil)
+	os.Rename(traefikTestAccessLogFile, traefikTestAccessLogFile+".rotated")
+
+	//continue issuing requests
+	_, err = http.Get("http://127.0.0.1:8000/test2")
+	c.Assert(err, checker.IsNil)
+	_, err = http.Get("http://127.0.0.1:8000/test2")
+	c.Assert(err, checker.IsNil)
+
+	// Verify access.log.rotated output as expected
+	rotated, err := os.Open(traefikTestAccessLogFile + ".rotated")
+	c.Assert(err, checker.IsNil)
+	rotatedLog := bufio.NewScanner(rotated)
+	count := 0
+	for rotatedLog.Scan() {
+		line := rotatedLog.Text()
+		c.Log("rl: " + line)
+		if len(line) > 0 {
+			CheckAccessLogFormat(c, line, count)
+		}
+		count++
+	}
+	c.Assert(count, checker.Equals, 1)
+
+	// Verify access.log output as expected
```

Comment:
How about removing the duplicated block by introducing a loop over both log file names?